src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala

```diff
@@ -225,9 +225,21 @@ object DiplomaticObjectModelAddressing {
       grandParentOpt = b.device.get.parent
       gp <- grandParentOpt
     } yield OMInterrupt(
-            receiver = getDeviceName(gp),
-            numberAtReceiver = getInterruptNumber(b.value),
-            name = name
-          )
-    }
-}
+      receiver = getDeviceName(gp, resources),
+      numberAtReceiver = getInterruptNumber(b.value),
+      name = name
+    )
+  }
+
+  def describeGlobalInterrupts(name: String, resources: ResourceBindings): Seq[OMInterrupt] = {
+    val bindings = resources("int")
+    for {
+      binding <- bindings
+      device = binding.device.get
```

Comment:
Is `binding.device` an `Option`? I'm trying to infer based on the call to `.get` here. If it is, you probably want to write this like:

```scala
for {
  binding <- bindings
  device <- binding.device
} yield ...
```

The difference is that the way you wrote it, if the `binding.device` is `None`, it'll throw an exception because you're calling `.get` on it, but the way I wrote it, if it's a `None`, it'll simply skip over to the next `binding`, treating it essentially like an empty `Seq`.